### ISSUE TITLE
Custom EditorImportPlugin that saves ImageTexture, will update saved imported resource on disk, but Editor won't update resource without restart

### ISSUE THREAD

**Issue by RabbitB** [godotengine/godot#17258]
**Godot version:**
Godot 3.0.1 win64 Stable

**OS/device including version:**
Windows 10 Pro x64

**Issue description:**
As the (unfortunately long-winded) title states, a custom EditorImportPlugin that saves an ImageTexture, will manage to properly update the imported resource on disk, but the resource fails to update within the editor, and requires a complete restart of the editor before the file will update.

Other than an editor restart, nothing I've tried will force the editor to update properly. Automatic reimport by changing the source file; manual reimport, etc.

Also of note, upon reimport, this error is generated:

` scene\resources\texture.cpp:107 - Condition ' err != OK ' is true.`

This is the method that error is occuring in:

```C++
void ImageTexture::reload_from_file() {

    String path = ResourceLoader::path_remap(get_path());
    if (!path.is_resource_file())
        return;

    uint32_t flags = get_flags();
    Ref<Image> img;
    img.instance();

    Error err = ImageLoader::load_image(path, img);
    ERR_FAIL_COND(err != OK);

    create_from_image(img, flags);
}
```

It should be noted that this issue might occur for multiple different resource types. I've only tried two, `ImageTexture` and `SpatialMaterial`. Saving a `SpatialMaterial` has an issue with the resource preview updating upon reimport, but the actual resource updates properly in the editor (so changes can be seen propagating in scenes and other places where used).

**Steps to reproduce:**

This is a case where the included project will need to be run to verify everything that's happening, but I can include the relevant code here:

```gdscript
func import(source_file, save_path, options, r_platform_variants, r_gen_files):

	var img_gen_data = read_img_gen_file(source_file)
	
	if typeof(img_gen_data) != TYPE_DICTIONARY:
		return img_gen_data

	if options.use_red_anyway:
		img_gen_data.color = Color(255, 0, 0)

	var generated_image = Image.new()
	generated_image.create(img_gen_data.width, img_gen_data.height, false, Image.FORMAT_RGBA8)
	generated_image.lock()
	generated_image.fill(img_gen_data.color)
	generated_image.unlock()
	
	var image_texture = ImageTexture.new()
	image_texture.create_from_image(generated_image)

	return ResourceSaver.save("%s.%s" % [save_path, get_save_extension()], image_texture)
```

You can reproduce the error yourself by changing the included source file `sample.img_gen`, or by flagging the import option `use_red_anyway`, which will disregard the fill color from the file, and just use red. You'll be able to see that the resource doesn't update, both in the resource viewer, and in the included scene (which is just a `Sprite` in the center, that uses the generated `ImageTexture`).

**Minimal reproduction project:**
I've adapted this sample project from the EditorImportPlugin tutorial available in the docs.
[EditorImportPlugin_ImageTexture_CacheIssue.zip](https://github.com/godotengine/godot/files/1778746/EditorImportPlugin_ImageTexture_CacheIssue.zip)

**Comment by RabbitB** [comment]
I'm adding this just as an aside, not directly related to this issue itself, but could be related to how custom images are imported. `StreamTexture` as far as I can tell, has no way of being generated from GDScript. The methods for creating a new `StreamTexture` are never exposed to the editor, and I thought it strange that the built-in image importers convert to `StreamTexture`, but it can't be done within a custom importer.

**Comment by UgisBrekis** [comment]
Stumbled upon the same exact problem myself now, only in my case it's also with the AtlasTexture. All the generated textures appear in OS's file explorer as they should, but the editor only updates that after restart.

As an addition, AtlasTexture thumbnail in editor never get's updated. My guess is that the thumbnail was generated for the png first time only, because it was imported as a default texture. After I switch to my custom importplugin, thumbnail is simply not being generated, which lead to it being stuck in the same state forever.

**Comment by MightyPrinny** [comment]
This happened to me when I updated an atlas made with free texture packer in 3.1.2.

**Comment by KoBeWi** [comment]
Seems fixed in 3.2.4 beta2 (the scene part)
Preview not refreshing is tracked here #33745
